extension/src/plots/model/collect.ts

```diff
@@ -502,49 +503,91 @@ export const collectTemplates = (data: PlotsOutput): TemplateAccumulator => {
 }
 
 const updateDatapoints = (
-  datapoints: unknown[],
+  path: string,
+  revisionData: RevisionData,
+  selectedRevisions: string[],
   key: string,
   fields: string[]
 ): unknown[] =>
-  datapoints.map(data => {
-    const obj = data as Record<string, unknown>
-    return {
-      ...obj,
-      [key]: mergeFields(fields.map(field => obj[field] as string))
+  selectedRevisions
+    .flatMap(revision =>
+      revisionData?.[revision]?.[path].map(data => {
+        const obj = data as Record<string, unknown>
+        return {
+          ...obj,
+          [key]: mergeFields(fields.map(field => obj[field] as string))
+        }
+      })
+    )
+    .filter(Boolean)
+
+const updateRevisions = (
+  selectedRevisions: string[],
+  domain: string[]
+): string[] => {
+  const revisions: string[] = []
+  for (const revision of selectedRevisions) {
+    for (const entry of domain) {
+      revisions.push(mergeFields([revision, entry]))
     }
-  })
+  }
+  return revisions
+}
 
-const stringifyDatapoints = (
-  datapoints: unknown[],
-  field: string | undefined,
-  isMultiView: boolean
-): string => {
+const transformRevisionData = (
```

Comment:
[F] This change is good because we are not looping over the datapoints twice anymore.